nexus/db-queries/src/db/queries/network_interface.rs

```diff
@@ -386,6 +392,12 @@ fn decode_database_error(
                 let mac = interface.mac.unwrap_or_else(|| MacAddr::from_i64(0));
                 InsertError::MacAddressNotAvailable(mac)
             }
+            // Constraint violated if a user-requested slot has
+            // already been assigned within the same instance or service.
+            Some(constraint) if constraint == SLOT_NOT_AVAILABLE_CONSTRAINT => {
+                let slot = interface.slot.unwrap_or(0);
```

Comment:
Do we care about the case where the slot is `None` here? That shouldn't be possible, since we'll only get this error if the user requested one, meaning slot is `Some(_)`.

Reply:
Agreed, and I think we came to the same conclusion with `MAC_NOT_AVAILABLE_CONSTRAINT` (which is immediately above this new case). We could be more pedantic and store the option instead of having to `unwrap_or()`, but I'm not sure it's really useful.